biggerquery/beam_manager.py

```diff
@@ -0,0 +1,205 @@
+        self.region = region
+        self.machine_type = machine_type
+        self.standard_sql = True
+
+    def read_from_big_query(self, sql):
+        return self.beam_manager.read_from_big_query(
+            self.dataset,
+            self.project_id,
+            self.standard_sql,
+            sql
+        )
+
+    def read_from_avro(self, file_source_path):
+        return self.beam_manager.read_from_avro(file_source_path)
+
+    def write_to_avro(self, file_output_path, schema):
+        return self.beam_manager.write_to_avro(file_output_path, schema)
+
+    @property
+    def run_datetime(self):
+        return self.beam_manager.run_datetime
+
+    def write_truncate_to_big_query(self, table_name, schema):
+        fields = [bigquery.TableFieldSchema(name=row['name'], type=row['type'].upper(), mode=row['mode'].upper()) for row in schema]
+        schema = bigquery.TableSchema(fields=fields)
+        return self.beam_manager.write_truncate_to_big_query(table_name, schema)
+
+    def create_dataflow_pipeline(self,
+                                 job_name,
+                                 local_runner=None):
+        options = PipelineOptions()
+
+        if not local_runner:
+            google_cloud_options = options.view_as(GoogleCloudOptions)
+            google_cloud_options.project = self.project_id
+            google_cloud_options.job_name = self._job_name(self.run_datetime, job_name)
+            google_cloud_options.staging_location = 'gs://{dataflow_bucket}/beam_runner/staging'.format(
+                dataflow_bucket=self.dataflow_bucket)
+            google_cloud_options.temp_location = 'gs://{dataflow_bucket}/beam_runner/temp'.format(
+                dataflow_bucket=self.dataflow_bucket)
+            google_cloud_options.region = self.region
+            options.view_as(WorkerOptions).machine_type = self.machine_type
+
+            options.view_as(SetupOptions).requirements_file = self.requirements_file_path
+            options.view_as(StandardOptions).runner = 'DataflowRunner'
+        else:
+            options.view_as(StandardOptions).runner = 'DirectRunner'
+        self.beam_manager.beam_manager.pipeline = beam.Pipeline(options=options)
+        return self.beam_manager.beam_manager.pipeline
+
+    def _job_name(self, day, job_name):
+        if 'job_suffix' in self.beam_manager.extras:
+            import re
+            table_prefix_slug = re.sub(r'[^a-z0-9]+', '-', self.beam_manager.extras['job_suffix']).strip('-')
+            return '{job_name}-{day}-{job_suffix}'.format(job_name=job_name, day=day, job_suffix=("" if table_prefix_slug is None else table_prefix_slug))
+        else:
+            return '{job_name}-{day}'.format(job_name=job_name, day=day)
+
+
+def _throw_on_none(param, param_name):
+    if param is None:
+        raise ValueError("%s must be defined" % param_name)
+
+
+def create_dataflow_manager(
+        project_id,
+        runtime,
+        dataset_name,
+        dataflow_bucket,
```

Comment:
internal_tables and external_tables are optional so they should be None by default. I think that the machine_type parameter should has default value to(standard machine).

Reply:
external_tables is optional, right. But without internal_tables it doesnt make sense to use this tool

Reply:
Look now please.

Reply:
You can write full table names without using aliases, so it make sense.

Reply:
Oh, right

Reply:
Look now please

Reply:
Look now please